pkg/apis/core/types.go

```diff
@@ -2175,6 +2175,41 @@ const (
 	PreemptNever PreemptionPolicy = "Never"
 )
 
+// DisruptionPolicy specifies how pods with this priority class may be disrupted when being preempted.
+type DisruptionPolicy struct {
```

Comment:
This is significantly different from what was agreed on in the KEP

https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens#design-details

I'm ok having a struct, but just with one field PriorityGreaterThanOrEqual

Reply:
When implementing this feature, I found that just one field PriorityGreaterThanOrEqual can't cover all situations. There are three situations listed below:

```go
const (
	// PDBIgnore indicates that the associated PDB will not be considered
	// when this Pod is preempted during scheduling.
	PDBIgnore PDBDisruptionPolicy = "PDBIgnore"
	// PDBBestEffort indicates that the associated PDB should be protected as
	// much as possible from being violated when this Pod is preempted during scheduling.
	PDBBestEffort PDBDisruptionPolicy = "PDBBestEffort"
	// PDBStrict indicates that the associated PDB should be protected strictly from
	// being violated when this Pod is preempted during scheduling.
	// But if the preempter has a priority class name of `system-cluster-critical` or
	// `system-node-critical`, may still potentially violate the victim's PDB.
	PDBStrict PDBDisruptionPolicy = "PDBStrict"
)
```

We may also need to provide an option for the user that the scheduler ignores the PDB when choosing the victims during scheduling. This allows users to make their choices clear and predictable. WDYT?

Reply:
1. A behavior like `PDBIgnore` was not part of the goals and I don't see why it's useful.
2. `PDBStrict` is basically the same as `PDBBestEffort` with `priorityGreaterThanOrEqual=value(system-node-critical)`

So I don't think it's useful.